src/m365/spo/commands/folder/folder-get.ts

```diff
@@ -108,6 +113,15 @@ class SpoFolderGetCommand extends SpoCommand {
 
     try {
       const folder = await request.get<FolderProperties>(requestOptions);
+      if (args.options.withPermissions) {
+        const listItemAllFields = folder.ListItemAllFields;
+        if (!(listItemAllFields ?? false)) {
+          throw Error('Please ensure the specified folder URL or folder Id does not refer to a root folder. Use \'spo list get\' with withPermissions instead.');
+        }
+        listItemAllFields.RoleAssignments.forEach(r => {
+          r.Member.PrincipalTypeString = ListPrincipalType[r.Member.PrincipalType];
+        });
+      }
```

Comment:
the problem is that part of the response is actually not readable for a human being 😋
```json
 {
            "BasePermissions": {
              "High": "48",
              "Low": "134287360"
            },
            "Description": "Can view specific lists, document libraries, list items, folders, or documents when given permissions.",
            "Hidden": true,
            "Id": 1073741825,
            "Name": "Limited Access",
            "Order": 160,
            "RoleTypeKind": 1
          }
```

we do not know what `BasePermissions` and `RoleTypeKind` is. So in order to address this in CLI we extend the response with `BasePermissionsValue` and `RoleTypeKindValue` to translate those numbers to meaningful values. 
Please refer to [roledefinition-list](https://github.com/pnp/cli-microsoft365/blob/main/src/m365/spo/commands/roledefinition/roledefinition-list.ts) to see how we may address this 👍